sdk/core/core-http/lib/policies/proxyPolicy.ts

```diff
@@ -13,6 +13,32 @@ import { WebResource } from "../webResource";
 import { Constants } from "../util/constants";
 import { URLBuilder } from "../url";
 
+/**
+ * Options to configure a proxy for outgoing requests.
+ */
+
+export interface ProxyOptions {
```

Comment:
We also have `ProxySettings` which follows the same shape as the new `ProxyOptions`. If we like the name `ProxyOptions`, then perhaps stop exporting `ProxySettings`?

Reply:
Removing `ProxySettings` would be a breaking change for anyone using `ServiceClientOptions` (`arm-*` library users, basically), so I added a different `ProxyOptions` that we can change independently.  I made it look exactly like `ProxySettings` mainly so that they could be used interchangeably, but I think based on your other comment about proxy URL they'll end up deviating a bit anyway.  I also want to rename `username` to `userName` :)

Reply:
> Removing ProxySettings would be a breaking change for anyone using ServiceClientOptions (arm-* library users, basically)

The arm-* libraries dont use core-http yet.
By the time they do, we would have gotten rid of `ServiceClientOptions` itself from `core-http` and have the code generator use the new `ProxyOptions`
So as long as we are saying that we keep `ProxySettings` around as long as `ServiceClientOptions` is around, I guess am ok with that.

Reply:
Yep, `ProxySettings` will go away with `ServiceClientOptions`.